app/assets/javascripts/beak/widgets/ractives/code-editor.coffee

```diff
@@ -78,16 +83,68 @@ window.RactiveModelCodeComponent = Ractive.extend({
     keywordsList = Array.from(keywords).map((item) -> item.replace("\\", ""))
     Object.keys(@getProcedureNames()).concat(keywordsList)
 
+  setupCodeUsagePopup: ->
+    editor = @findComponent('codeEditor').getEditor()
+    codeUsageMap = {
+      'Ctrl-U': =>
+        if editor.somethingSelected()
+          @setCodeUsage()
+      ,'Cmd-U': =>
+        if editor.somethingSelected()
+          @setCodeUsage()
+    }
+
+    editor.on('cursorActivity', (cm) =>
+      if @get('usageVisibility')
+        @set('usageVisibility', false)
+    )
+    
+    return
+
+  getCodeUsage: ->
+    editor = @findComponent('codeEditor').getEditor()
+    selectedCode = editor.getSelection().trim()
+    @set('selectedCode', selectedCode)
+    codeString = @get('code')
+    check = ///\b(#{selectedCode})\b///g
+    codeUsage = []
+    while (match = check.exec(codeString))
+      pos        = editor.posFromIndex(match.index + match[1].length)
+      lineNumber = pos.line + 1
+      line       = editor.getLine(pos.line)
+      codeUsage.push( { pos, lineNumber, line } )
+    codeUsage
+      
+  setCodeUsage: ->
+    codeUsage = @getCodeUsage()
+    editor = @findComponent('codeEditor').getEditor()
+    @set('codeUsage', codeUsage)
+    pos = editor.cursorCoords(editor.getCursor())
+    @set('usageLeft', pos.left)
+    @set('usageTop', pos.top)
+    @set('usageVisibility', true)
+    return
+
   on: {
     'complete': (_) ->
       @setupProceduresDropdown()
       CodeMirror.registerHelper('hint', 'fromList', @netLogoHintHelper)
       @setupAutoComplete(@autoCompleteWords())
+      @setupCodeUsagePopup()
       return
 
     'recompile': (_) ->
       @setupAutoComplete(@autoCompleteWords())
       return
+
+    'jumpToUsage': (context, usagePos) ->
```

Comment:
Sorry, should've spotted this before, but our preferred event naming style would be hyphen-separated terms, so `'jump-to-usage'` in this case.